### ISSUE TITLE
Cypress + VPN + MacOS + Private Relay

### ISSUE THREAD

**Issue by s-chathuranga-j** [cypress-io/cypress#30315]
### Current behavior

Cypress tests that depend on VPN access may fail on MacOS systems, even when the VPN is correctly configured and active. This occurs because Cypress is unable to route its traffic through the VPN, resulting in failed connections or inaccessible domains. But if you try to access those domains with a regular browser on that laptop, it works. Only accessing those domains via Cypress execution is failing.

### Desired behavior

**The reason in such cases is iCloud+ Private Relay is turned ON on that MacOS system.** Cypress fails to utilize the VPN connection when this feature is turned on. It's an important feature in Apple eco system. Since the normal browser can utilize this I hope this can be fixed for Cypress execution too.

### Test code to reproduce

Write a test for domain which needs a VPN connection to access the domains

### Cypress Version

13.14.2

### Node version

20.12.2

### Operating System

macOS 15.1 Beta

**Comment by jennifer-shehane** [comment]
@s-chathuranga-j Can you disable Private Relay for Cypress / the traffic needed within tests. We proxy all requests, so there's likely information that we need that is obscured in Private Relay - I'm not sure our app could even function with this being ignored.

**Comment by s-chathuranga-j** [comment]
@jennifer-shehane Nope, there is no way in MacOS to omit specific apps from utilizing this feature. But the question is the normal browser (non cypress launched), utilize the VPN over Private Relay. ONLY cypress launched browser have this issue.

**Comment by jennifer-shehane** [comment]
@s-chathuranga-j We have specific proxy logic, so that you can intercept requests and change the repsonses for example, so we are accessing this for legitimate use cases. I'm not sure if there's a way around this, we'd have to investigate. We're not currently prioritizing looking into this however.
